### DISCUSSION TITLE
ERROR: language "plcsharp" does not exist after installation

### DISCUSSION THREAD

**Discussion by ADRNV** [Brick-Abode/pldotnet#10] [Q&A] [answered]
Hi i install PL/.NET on my WSL machine.

My WSL setup:
OS: Ubuntu 22.04
Kernel:  x86_64 Linux 5.15.146.1-microsoft-standard-WSL2
CPU: AMD Ryzen 5 3500U
RAM: 3446MiB

1. I install `dotnet 6.0(sdk, runtime, hostfrx), libglib2.0`. Postgres 14.11 was installed
2. I install .deb package for my Postgres version (postgresql-**14**-pldotnet_0.99-rc1_amd64.deb)
3. I try to test PL/.NET - execute `do`  from manual from site block with plcsharp from psql client, but i get this error - 'language "plcsharp" does not exist'
4. I tryed restart postgres service, but error still same

Thanks for help

**Comment by tlewis-ba** [comment, marked as answer]
This is a common error; it means that pl/dotnet was not initialized properly inside of PostgreSQL.

It might be an error in our dpkg; we will need to test it.

Can you try running the initialization script manually and posting the results here?  It should be located in `/usr/share/postgresql/14/extension/pldotnet--0.9.sql`.

**Reply by ADRNV** [reply]
"It might be an error in our dpkg; we will need to test it" - output after installation of dpkg:
![image](https://github.com/Brick-Abode/pldotnet/assets/49834917/f4d7163b-a1ee-4e80-b8fb-64e4683cbc15)
Result when run from `CREATE EXTENSION` in Postgres:
```
create extension pldotnet
[2024-05-01 15:39:28] [58P01] ERROR: could not load library "/usr/lib/postgresql/14/lib/pldotnet.so": libnethost.so: cannot open shared object file: No such file or directory
```
Thats a... magic... **after** **reinstall** dpkg package extension insalles in Postgres, and execute .NET without errors
```
[2024-05-01 15:44:45] completed in 525 ms
weak_struct_data_test.public> DO $$ Elog.Info("Hello PL.NET!"); $$ LANGUAGE plcsharp
Hello PL.NET!
```

**Reply by ADRNV** [reply]
*package was installed from site https://pldotnet.brickabode.com/how-to-install-pl-net in all cases

**Reply by tlewis-ba** [reply]
1. You got it working; great!
2. As you play around with it, feel free to post your experiences here, especially the Really Cool stuff.
3. It's weird that it failed the first time and succeeded the second time.  Not sure how to debug that, but it is definitely something we will keep an eye on and try to test more widely in our subsequent releases.

I appreciate the report and am really glad you got it working.

**Reply by ADRNV** [reply]
Yeah im same glad. Thanks for support. I have some use cases for this extension. Thats real cool extenstion, no need to use Java) I close this discussion ?

**Reply by tlewis-ba** [reply]
Yes, please close, and feel free to open a new one in case of new problems, or to share with us the (java-free) awesomeness you use pl/dotnet to create in the universe!
